lib/nuget-parser/index.ts

```diff
@@ -75,32 +75,51 @@ function getFileContents(fileContentPath: string): string {
   }
 }
 
+function tryToGetFileByName(dir: string, filename: string): null | Buffer {
+  const depsFilePath = path.join(dir, filename);
+  try {
+    const depsFile = fs.readFileSync(depsFilePath);
+    if (depsFile) return depsFile;
+  } catch (_) {
+    // Due to race conditions, fs docs suggests to not use .stat or .access to check if a file exists
+    // but instead we should to try and read it.
+    // https://nodejs.org/api/fs.html#fsstatpath-options-callback
+  }
+  return null;
+}
+
 // `dotnet` can publish the .deps file to a variety of places inside the publish folder, depending on what you're
 // including and targeting. Instead of trying different directories, just scan them all. In most cases, the file
 // will be in the root directory. (See https://github.com/Azure/azure-functions-vs-build-sdk/issues/518)
-function findDepsFilePathInPublishDir(dir: string): string | null {
+function findDepsFileInPublishDir(dir: string, filename): Buffer | null {
+  let renamedFile: Buffer | null = null;
+
+  // Try to get the file via full path.
+  const namedFile = tryToGetFileByName(dir, filename);
+  if (namedFile) return namedFile;
+
   for (const item of fs.readdirSync(dir)) {
     const itemPath = path.join(dir, item);
 
     // The file is usually <project>.deps.json, but in edge cases, `dotnet` names it for you.
     if (itemPath.endsWith('deps.json')) {
-      return itemPath;
+      renamedFile = fs.readFileSync(itemPath);
     }
 
     if (!fs.statSync(itemPath).isDirectory()) {
       continue;
     }
 
     // Otherwise, look in a nested dir for the same thing.
-    const foundFilePath = findDepsFilePathInPublishDir(itemPath);
-    if (!foundFilePath) {
+    const foundFile = findDepsFileInPublishDir(itemPath, filename);
```

Comment:
This will return one of the three: namedFile, renamedFile or null. At this point in here we're going to make a confusion between namedFile and renamedFile and exit faster in case of renamedFile, but I think it's safe, could not found cases of nested deps.json files with different names. If such cases are to exist tho', we will catch them with the same error as we caught this one.